agent/structs/config_entry_gateways.go

```diff
@@ -231,11 +239,49 @@ func (e *IngressGatewayConfigEntry) validateServiceSDS(lis IngressListener, svc
 	return nil
 }
 
+func validateGatewayTLSConfig(tlsCfg GatewayTLSConfig) error {
+	if tlsCfg.TLSMinVersion != types.TLSVersionUnspecified {
+		if err := types.ValidateTLSVersion(tlsCfg.TLSMinVersion); err != nil {
+			return err
+		}
+	}
+
+	if tlsCfg.TLSMaxVersion != types.TLSVersionUnspecified {
+		if err := types.ValidateTLSVersion(tlsCfg.TLSMaxVersion); err != nil {
+			return err
+		}
+	}
+
+	// TODO: Is there any reasonable way to validate that TLSMaxVersion is
```

Comment:
You would have to write an explicit comparison, possibly assigning these strings an ordinal number that you can use for relative comparisons if that is what you want.

Reply:
That's basically the entire approach I had started with and then wound up unwinding in favor of passing string types around everywhere, wouldn't be particularly excited about re-implementing all that again 😞

Reply:
Welp, ended up writing a short comparator function just to handle less than checks for concrete values (won't cover implementation-dependent values such as when either side is unspecified or explicitly set to `TLS_AUTO`)